codedigger/lists/tests/test_views.py

```diff
@@ -0,0 +1,44 @@
+from .test_setup import TestSetUp
+from user.models import User, Profile
+from django.urls import reverse
+from rest_framework.test import APIClient
+from lists.models import List,ListInfo, Solved
+from problem.models import Problem
+
+
+
+class TestViews(TestSetUp):
+
+    #anon checking of list of all ladders and list endpoints
+    def test_check_topicwise_list_all_lists_view(self):
+        test_url = reverse('topicwise-list')
+        res = self.client.get(test_url)
+        self.assertEqual(res.status_code, 200)
+        
+    def test_check_topicwise_ladder_all_ladders_view(self):
+        test_url = reverse('topicwise-ladder')
+        res = self.client.get(test_url)
+        self.assertEqual(res.status_code, 200)
+        
+    def test_check_levelwise_list_all_lists_view(self):
+        test_url = reverse('levelwise-list')
+        res = self.client.get(test_url)
+        self.assertEqual(res.status_code, 200)
+    
+    def test_check_levelwise_ladder_all_ladder_view(self):
+        test_url = reverse('levelwise-ladder')
+        res = self.client.get(test_url)
+        self.assertEqual(res.status_code, 200)
+   
+   #same as above but with an authenticated user 
+    def test_auth_check_topicwise_list_all_lists_view(self):
+        test_url = "http://127.0.0.1:8000/lists/topicwise/list/testinglist_topicwise"
```

Comment:
Use reverse functions.

Reply:
so reverse('topicwise-list') + list_name ?

Reply:
Don't know, Have to go through documentation.